packages/salesforcedx-vscode-apex/src/commands/forceGenerateFauxClasses.ts

```diff
@@ -191,16 +191,52 @@ export async function initSObjectDefinitions(projectPath: string) {
   const hasDefaultUsernameSet =
     (await getDefaultUsernameOrAlias()) !== undefined;
   if (projectPath && hasDefaultUsernameSet) {
-    const sobjectFolder = path.join(
-      projectPath,
-      SFDX_DIR,
-      TOOLS_DIR,
-      SOBJECTS_DIR
-    );
+    const sobjectFolder = getSObjectsDirectory(projectPath);
     if (!fs.existsSync(sobjectFolder)) {
       forceGenerateFauxClassesCreate(SObjectRefreshSource.Startup).catch(e => {
         throw e;
       });
     }
   }
 }
+
+function getSObjectsDirectory(projectPath: string) {
+  return path.join(
+    projectPath,
+    SFDX_DIR,
+    TOOLS_DIR,
+    SOBJECTS_DIR
+  );
+}
+
+export async function checkSObjectsAndRefresh(projectPath: string) {
+  if (projectPath) {
+    if (!fs.existsSync(getSObjectsDirectory(projectPath))) {
+      telemetryService.sendEventData('sObjectsRefresh',
+        { type: 'No SObjects' },
+        undefined);
+      const message = nls.localize('sobjects_refresh_needed');
+      const buttonTxt = nls.localize('sobjects_refresh_now');
+      const shouldRefreshNow = await notificationService.showInformationMessage(message, buttonTxt);
+      if (shouldRefreshNow && shouldRefreshNow === buttonTxt) {
+        telemetryService.sendEventData('sObjectsRefresh',
```

Comment:
This is a nit but can we name the event something like sObjectRefreshPrompt or sObjectRefreshNotification so that we know it's related to the notification? Sometimes the understanding of certain events based off their names can get lost after awhile.